librustls/src/client.rs

```diff
@@ -770,14 +770,11 @@ impl rustls_client_config {
             };
             let config = try_clone_arc!(config);
             let conn_out = try_mut_from_ptr_ptr!(conn_out);
-            let server_name = match server_name.to_str() {
-                Ok(s) => s,
-                Err(std::str::Utf8Error { .. }) => return rustls_result::InvalidDnsNameError,
-            };
-            let server_name = match server_name.try_into() {
-                Ok(sn) => sn,
-                Err(_) => return rustls_result::InvalidDnsNameError,
+
+            let Ok(server_name) = server_name.to_str() else {
+                return rustls_result::InvalidDnsNameError;
             };
+            let Ok(server_name) = server_name.try_into() else { return rustls_result::InvalidDnsNameError; }
```

Comment:
Nit: no rustfmt because macro? I think this should get a similar style as the `server_name` one.

Reply:
We do [a special `sed` dance](https://github.com/cpu/rustls-ffi/blob/4cf5a4b77065fb48ff18c06322a31414f667e2b8/librustls/cmake/format.cmake#L1-L7) in this repo (and `rustls-openssl-compat`) to make rustfmt work inside the macros.

It's easy to forget :'( In this case it looks like I probably ran vanilla `cargo fmt` and then later `cmake --target format-fix` with the `sed` version.

All fixed!